### PR TITLE
OSD-7608 validation of image digest and version mentioned in UpgradeC…

### PR DESCRIPTION
**Pull request by a7vicky**
### What type of PR is this?
Feature

### What this PR does / why we need it?
MUO should throw validation error before starting upgrade that the version from .spec.desired.version and from the .spec.desired.image version are not same.

### Which Jira/Github issue(s) this PR fixes?
https://issues.redhat.com/browse/OSD-7608

### Pre-checks (if applicable):
- [x] Tested latest changes against a cluster
- [ ] Ran `make generate` command locally to validate code changes
- [ ] Included documentation changes with PR

### CODE REVIEW

File: pkg/validation/validation.go [hunk 1]
```diff
@@ -316,3 +348,85 @@ func supportsVersionUpgrade(uc *upgradev1alpha1.UpgradeConfig) bool {
...
+	client := http.Client{
+		Timeout: time.Second * 20,
+	}
+
+	res, getErr := client.Do(req)
+	if getErr != nil {
+		return nil, getErr
+	}
+
+	if res.Status != "200 OK" {
```
Comment (mrbarge) [inline pkg/validation/validation.go:411]: Better to compare `res.StatusCode` to `http.StatusOK`.

Comment (a7vicky) [reply pkg/validation/validation.go]: @mrbarge Thank you for your review. Applied the suggested change.

File: pkg/validation/validation.go [hunk 2]
```diff
@@ -316,3 +348,85 @@ func supportsVersionUpgrade(uc *upgradev1alpha1.UpgradeConfig) bool {
...
+	ref, _ := imagereference.Parse(image)
+	manifesturl := url.URL{
+		Scheme: "https",
+		Host:   ref.Registry,
+		Path:   "v2" + "/" + ref.Namespace + "/" + ref.Name + "/" + "manifests" + "/" + ref.ID,
+	}
+
+	body, err := runHTTP(manifesturl.String())
+	if len(body) == 0 {
+		return "", fmt.Errorf("Failed to fetch image manifest: %s  may not be a valid image", image)
```
Comment (ravitri) [inline pkg/validation/validation.go:363]:
```suggestion
		return "", fmt.Errorf("Failed to fetch image manifest digest: %s may not be a valid release image", image)
```

File: pkg/validation/validation.go [hunk 3]
```diff
@@ -316,3 +348,85 @@ func supportsVersionUpgrade(uc *upgradev1alpha1.UpgradeConfig) bool {
...
+	}
+	bloburl := url.URL{
+		Scheme: "https",
+		Host:   ref.Registry,
+		Path:   "v2" + "/" + ref.Namespace + "/" + ref.Name + "/" + "blobs" + "/" + manifest.Config.Digest,
+	}
+
+	resbody, err := runHTTP(bloburl.String())
+	if len(resbody) == 0 {
+		return "", fmt.Errorf("Failed to fetch image imageconfig for %s", image)
```
Comment (ravitri) [inline pkg/validation/validation.go:381]:
```suggestion
		return "", fmt.Errorf("Failed to fetch blobs for image manifest digest: %s needs to be a valid release image", image)
```

**Comment by ravitri** [comment]
Tested the changes and seems fine to me overall. Thanks @a7vicky ! 🎉 
/lgtm

Would need an approver to approve the changes ahead after review.

**Comment by mrbarge** [comment]
Tested this also, it's good - let's bring it in.
/approve

**Pull request merged**